app/src/main/java/com/spascoding/taskycourse/feature_agenda/presentation/detail_screen/DetailScreen.kt

```diff
@@ -0,0 +1,151 @@
+import com.spascoding.taskycourse.core.presentation.components.TaskyScaffold
+import com.spascoding.taskycourse.feature_agenda.presentation.utils.DATE_FORMAT
+import com.spascoding.taskycourse.navigation.Navigation
+
+@Composable
+fun DetailScreenRoot(
+    navController: NavController,
+    navigation: Navigation,
+    viewModel: DetailViewModel = hiltViewModel(),
+) {
+
+    val state by viewModel.state.collectAsStateWithLifecycle()
+
+    val onEvent: (DetailEvent) -> Unit = { event ->
+        when (event) {
+            else -> viewModel.onEvent(event)
+        }
+    }
+
+    DetailScreen(
+        navController = navController,
+        navigation = navigation,
+        state = state,
+        onEvent = onEvent,
+    )
+}
+
+@Composable
+private fun DetailScreen(
+    navController: NavController,
+    navigation: Navigation,
+    state: DetailViewModelState,
+    onEvent: (DetailEvent) -> Unit,
+) {
+    TaskyScaffold(
+        topBar = {
+            HeaderSection(
+                navController = navController,
+                state = state,
+                onEvent = onEvent,
+            )
+        }
+    ) {
+        TitleSection()
+        DescriptionSection()
+        if(navigation is Navigation.EventDetailNavigation) {
+            PhotoSection()
+        }
+        StartDateSection()
+        if(navigation is Navigation.EventDetailNavigation) {
+            EndDateSection()
+        }
+        ReminderSection()
+        if(navigation is Navigation.EventDetailNavigation) {
+            AttendeeSection()
+        }
+        DeleteButton()
+    }
+}
+
+@Composable
+fun HeaderSection(
+    navController: NavController,
+    state: DetailViewModelState,
+    onEvent: (DetailEvent) -> Unit,
+) {
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Button(onClick = { navController.popBackStack() }) {
+            Icon(imageVector = Icons.Outlined.Close, contentDescription = null)
+        }
+        Text(
+            modifier = Modifier.weight(1f),
+            text = state.selectedDate.format(DATE_FORMAT).uppercase(),
+            textAlign = TextAlign.Center,
+            fontSize = 16.sp,
+        )
+        if (state.isEditMode) {
+            Button(onClick = { onEvent.invoke(DetailEvent.SaveAction) }) {
+                Text(text = stringResource(R.string.save))
+            }
+        } else {
+            Button(onClick = { onEvent.invoke(DetailEvent.EditAction) }) {
+                Icon(imageVector = Icons.Outlined.Edit, contentDescription = null)
+            }
+        }
+    }
+}
+
+@Composable
+fun TitleSection() {
+
+}
+
+@Composable
+fun DescriptionSection() {
+
+}
+
+@Composable
+fun PhotoSection() {
+
+}
+
+@Composable
+fun StartDateSection() {
+
+}
+
+@Composable
+fun EndDateSection() {
+
+}
+
+@Composable
+fun ReminderSection() {
+
+}
+
+@Composable
+fun AttendeeSection() {
+
+}
+
+@Composable
+fun DeleteButton() {
+
+}
```

Comment:
The screen will grow quite a lot, I'd keep all those composables in separate files, so you don't need to search forever to find something